### PR TITLE
Added audiopatch improvements

### PR DESCRIPTION
**Pull request by Wardergrip**
Audiopatch has more customisability for trailing audio sources and allows for limiting both amount of SFX played at the same time as well as a cooldown for playing.

### CODE REVIEW

File: ThatDefaultProject/Assets/ThatDefaultProject/Scripts/AudioPatch.cs [hunk 1]
```diff
@@ -1,107 +1,189 @@
...
+				return _longestClipTime;
+			}
+		}
+
+		private readonly List<PlayTimestamp> _playTimestamps = new();
+
+		private void Awake()
+		{
+			_uniqueClips = _audioClips.ToList(); //converting the given clips to a list, then putting them in _uniqueclips too
+			_longestClipTime = _uniqueClips.Max(x => x.length);
```
Comment (AtlantiaKing) [inline ThatDefaultProject/Assets/ThatDefaultProject/Scripts/AudioPatch.cs]: Not necessary to initialize _longestClipTime as it initialized itself

File: ThatDefaultProject/Assets/ThatDefaultProject/Scripts/AudioPatch.cs [hunk 2]
```diff
@@ -1,107 +1,189 @@
...
+			else
+			{
+				obj = Instantiate(_trailingAudioSourcePrefab);
+			}
+			DestroyTimer destroyTimer = obj.gameObject.AddComponent<DestroyTimer>();
+			destroyTimer.Time = _longestClipTime;
+			destroyTimer.StartTimer();
+			obj.transform.SetLocalPositionAndRotation(transform.position, transform.rotation);
+			obj.playOnAwake = false;
+			obj.rolloffMode = AudioRolloffMode.Linear;
```
Comment (AtlantiaKing) [inline ThatDefaultProject/Assets/ThatDefaultProject/Scripts/AudioPatch.cs]: Maybe make the rolloff mode customizable?

**Comment by Wardergrip** [comment]
Need to perform checks in Awake to avoid errors when you create a new patch.
Custom icon might be nice.
`ITransform` or something as parameter might be nice as an alternative?

**Comment by Wardergrip** [comment]
Added transform interface support and awake checks.

**Review by AtlantiaKing — APPROVED** [review]

**Pull request merged**